src/webgl/p5.RendererGL.js

```diff
@@ -1724,52 +1726,26 @@ p5.RendererGL = class RendererGL extends Renderer {
   }
 
   /*
-   * selects which fill shader should be used based on renderer state,
-   * for use with begin/endShape and immediate vertex mode.
+   * This method will handle both image shaders and
+   * fill shaders, returning the appropriate shader
+   * depending on the current context (image or shape).
    */
-  _getImmediateFillShader() {
-    const fill = this.userFillShader;
-    if (this._useNormalMaterial) {
-      if (!fill || !fill.isNormalShader()) {
-        return this._getNormalShader();
-      }
+   _getFillShader() {
+    if (this._drawingImage) {
+      return this.userImageShader;
```

Comment:
we could probably check if this.userImageShader is defined. If it's not defined we could return this._getLightShader() 

something like, 
```suggestion
if(this.userImageShader){
      return this.userImageShader;
      } else{
      return this._getLightShader();
      }
```